src/lib/gcode_simulator.py

```
@@ -0,0 +1,341 @@
+        self.bounds = Bounds()
+        self.debug = False
+        
+    def _parse_coord(self, line: str, current_pos: Point, current_feed) -> tuple[Point, float]:
+        new_pos = Point(current_pos.x, current_pos.y)
+        feed = current_feed
+        
+        x_match = re.search(r'X([-\d.]+)', line)
+        y_match = re.search(r'Y([-\d.]+)', line)
+        f_match = re.search(r'F([-\d.]+)', line)
+        
+        if x_match: new_pos.x = float(x_match.group(1))
```

Ruff (0.8.2)

Multiple statements on one line (colon)

(E701)